src/ch/ch_process.c

```diff
@@ -372,6 +374,95 @@ virCHProcessSetupVcpus(virDomainObjPtr vm)
     return 0;
 }
 
+/**
+ * chProcessNetworkPrepareDevices
+ */
+static int
+chProcessNetworkPrepareDevices(virCHDriverPtr driver, virDomainObjPtr vm)
+{
+    virDomainDefPtr def = vm->def;
+    size_t i;
+    virCHDomainObjPrivatePtr priv = vm->privateData;
+    g_autoptr(virConnect) conn = NULL;
+    char **tapfdName = NULL;
+    int *tapfd = NULL;
+    size_t tapfdSize = 0;
+
+    for (i = 0; i < def->nnets; i++) {
+        virDomainNetDefPtr net = def->nets[i];
+        virDomainNetType actualType;
+
+        /* If appropriate, grab a physical device from the configured
+         * network's pool of devices, or resolve bridge device name
+         * to the one defined in the network definition.
+         */
+        if (net->type == VIR_DOMAIN_NET_TYPE_NETWORK) {
+            if (!conn && !(conn = virGetConnectNetwork())){
+                return -1;}
```

Comment:
The brackets for this and the lower if seem out of place and not formatted correctly either way. Maybe just remove them?